app/models/doi.rb

```diff
@@ -674,82 +674,137 @@ def as_indexed_json(_options = {})
+          size: 10,
+          min_doc_count: 1,
+          include: "https?://orcid.org/.*"
         },
-        downloads: {
-          date_histogram: { field: "publication_year", interval: "year", format: "year", order: { _key: "desc" }, min_doc_count: 1 },
-          aggs: {
-            metric_count: { sum: { field: "download_count" } },
-            bucket_truncate: { bucket_sort: { size: 10 } },
+        aggs: {
+          creators_and_contributors: {
+            top_hits: {
+              _source: {
+                includes: [
+                  "creators_and_contributors.name",
+                  "creators_and_contributors.nameIdentifiers.nameIdentifier"
+                ]
+              },
+              size: 1
+            }
           },
+          "work_types": {
+            "terms": {
+              "field": "resource_type_id_and_name",
+              "min_doc_count": 1
+            }
+          }
+        }
+      },
+      views: {
+        date_histogram: { field: "publication_year", interval: "year", format: "year", order: { _key: "desc" }, min_doc_count: 1 },
+        aggs: {
+          metric_count: { sum: { field: "view_count" } },
+          bucket_truncate: { bucket_sort: { size: 10 } },
         },
-        citations: {
-          date_histogram: { field: "publication_year", interval: "year", format: "year", order: { _key: "desc" }, min_doc_count: 1 },
-          aggs: {
-            metric_count: { sum: { field: "citation_count" } },
-            bucket_truncate: { bucket_sort: { size: 10 } },
-          },
+      },
+      downloads: {
+        date_histogram: { field: "publication_year", interval: "year", format: "year", order: { _key: "desc" }, min_doc_count: 1 },
+        aggs: {
+          metric_count: { sum: { field: "download_count" } },
+          bucket_truncate: { bucket_sort: { size: 10 } },
         },
-      }
-    end
+      },
+      citations: {
+        date_histogram: { field: "publication_year", interval: "year", format: "year", order: { _key: "desc" }, min_doc_count: 1 },
+        aggs: {
+          metric_count: { sum: { field: "citation_count" } },
+          bucket_truncate: { bucket_sort: { size: 10 } },
+        },
+      },
+    }
+  end
+
+  def self.default_doi_query_facets
+    [
+      :resource_types,
+      :states,
+      :published,
+      :created,
+      :registered,
+      :providers,
+      :clients,
+      :affiliations,
+      :prefixes,
+      :schema_versions,
+      :link_checks_status,
+      :subjects,
+      :fields_of_science,
+      :licenses,
+      :certificates,
+      :views,
+      :downloads,
+      :citations
+    ]
+  end
+
+  def self.query_aggregations(disable_facets: false, facets: nil)
+    return {} if disable_facets
+
+    requested_facets = facets.respond_to?(:split) ? facets.split(",").map(&:strip).map(&:underscore).map(&:to_sym).uniq : default_doi_query_facets
+    requested_facets.index_with { |facet| all_doi_aggregations.dig(facet) }.compact
```

Comment:
I think this is similar to the slice method, available on hashes.
so this would become
`all_doi_aggregations.slice(*requested_facets)`
or
`AGGREGATION_DEFINITIONS .slice(requested_facets)`

keys that are not in the original hash are discarded.